### ISSUE TITLE
+required+ is not a valid predicate name (ArgumentError)

### ISSUE THREAD

**Issue by vyper** [hanami/validations#104]
# Gemfile

```ruby
source 'https://rubygems.org'

gem 'hanami-utils',       github: 'hanami/utils',       branch: 'master'
gem 'hanami-validations', github: 'hanami/validations', branch: 'master'
```

# testing.rb
```ruby
require 'hanami-validations'

class Testing
  include Hanami::Validations

  validations do
    required(:email) { format?(/@/) }
  end
end

result = Testing.new.validate
puts result.messages
```

But, when I ran command `bundle exec ruby testing.rb`:

```
/Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:195:in `[]': +required+ is not a valid predicate name (ArgumentError)
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:54:in `visit_predicate'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:18:in `visit'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:41:in `visit_val'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:18:in `visit'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:59:in `visit_and'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-logic-0.2.3/lib/dry/logic/rule_compiler.rb:18:in `visit'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:254:in `block in initialize_rules'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:253:in `each'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:253:in `each_with_object'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:253:in `initialize_rules'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:174:in `initialize'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:46:in `new'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/gems/dry-validation-0.7.4/lib/dry/validation/schema.rb:46:in `new'
	from /Users/vyper/.rvm/gems/ruby-2.3.0@hanami/bundler/gems/validations-34c34dfa416a/lib/hanami/validations.rb:57:in `validations'
	from testing.rb:6:in `<class:Testing>'
	from testing.rb:3:in `<main>'
```

**Comment by jodosha** [comment]
@vyper You should add the following lines to you `Gemfile:

```ruby
gem 'hanami-utils',   '~> 0.8', require: false, github: 'hanami/utils', branch: '0.8.x'
gem 'hanami-model',   '~> 0.7', require: false, github: 'hanami/model', branch: '0.7.x'
gem 'dry-types',                require: false, github: 'dry-rb/dry-types'
gem 'dry-logic',                require: false, github: 'dry-rb/dry-logic'
gem 'dry-validation',           require: false, github: 'dry-rb/dry-validation'
```

---

This may possibly not work yet, because of this regression: https://github.com/dry-rb/dry-validation/issues/183

**Comment by vyper** [comment]
@jodosha branch 0.6.x does not have the 34c34dfa416a7433df8226903f4a515200067246. I need to use master branch.

**Comment by jodosha** [comment]
@vyper I know. 😉  I'm not suggesting to use `0.6.x`, but to add the the gems listed in the (edited) `Gemfile` above. That would enable `0.7.x` series of `hanami-validations`.

**Comment by vyper** [comment]
But branch 0.7.x not exists @jodosha! d:

**Comment by jodosha** [comment]
@vyper you're right. Sorry for the misunderstanding but I'm jumping between GH issues, code and chat. It's a bit confusing.

Anyway, I can't merge 34c34df in `0.6.x` otherwise the build will break because of https://github.com/dry-rb/dry-validation/issues/183

---

Why do you need that behavior? I introduced for an experiment I'm doing with interactors.

**Comment by jodosha** [comment]
Fixed by #106 🎉

**Comment by vyper** [comment]
Thanks! (:
